### ISSUE TITLE
Attribute name change

### ISSUE THREAD

**Issue by CGastrell** [Automattic/crowdsignal-ui#127]
As seen on https://github.com/Automattic/crowdsignal-ui/blob/64c67fc06306dbaeacb6f60bb4cbdefb6ce1ec32/packages/block-editor/src/text-input/attributes.js#L6

that attribute should be called “question”. Otherwise we’ll need to parse the attributes different than `text-question` or do some mapping. The reason being there can’t be an input without a question (by design). If this needs to be optional, we can come up with the question as a sidebar entry or some other magic that can hold some default value. This is specially true for the results view where, without a question, responses to that question are somewhat confusing. Not to mention our results view can’t handle a question block without a question. 

What do you think @jcheringer @ice9js ?

**Comment by jcheringer** [comment]
Changing this `label` attribute to `question` seems a bit weird to me.
The `text-input` block is more generic and, from what I understood, it'll be mainly used to create form fields.
I wouldn't say that a form field has a `question`, seems more natural to me that it has a `label`.

**Comment by ice9js** [comment]
I think I agree with @jcheringer on this one.  

Calling things a certain way just because it works rather than what it actually is/does is confusing and leads to bad things. Plenty of examples around. This in particular is a very small thing, but in my experience, once you start taking shortcuts, no matter how little, it only gets worse as time passes on.

For parsing, considering the variety of blocks we want to implement and how that relates to the existing survey structure, I think the best thing we can do is be able to define maps per-block. Several blocks could be using the same map if they share the same attributes, but adding a new one should not be too different to how we just add another `attributes.js` on the frontend.

In terms of where we draw the line... Blocks describe the UI and thus should be aligned with that. And this might very well be different from what a _form_ is and how it works. So naturally, you would address these differences wherever the conversion happens which in our case is on the backend since we're sending raw editor output.  
And like I already said, we'll only be adding more blocks which may follow the same or a totally different structure so the problem would inevitably come back.

**Comment by ice9js** [comment]
>Now, let me ask again as I don't fully understand it: when using a text-input, where does the question come from? What is the responder answering/fulfilling?

Good point. I'd say that's still the `label` by default, however, @digitalwaveride might have another opinion on how this would work best.
